.github/workflows/codeql.yml

```diff
@@ -0,0 +1,78 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ 'main' ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ 'main' ]
+  schedule:
+    - cron: '35 14 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'python', 'cpp' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Use only 'java' to analyze code written in Java, Kotlin or both
+        # Use only 'javascript' to analyze code written in JavaScript, TypeScript or both
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    env:
+      SNAPCRAFT_BUILD_INFO: 1
+
+    steps:
+    - name: Install Snapcraft
+      uses: samuelmeuli/action-snapcraft@v2
+
+    - name: Install LXD
+      uses: canonical/setup-lxd@90d76101915da56a42a562ba766b1a77019242fd
```

Comment:
Could you please explain why this env var and these actions are included in this workflow?  I'm not seeing where they are being used in here.

Reply:
I saw them in other workflows like https://github.com/canonical/multipass/blob/main/.github/workflows/linux.yml. Please feel free to tweak the build!